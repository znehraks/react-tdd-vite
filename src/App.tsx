function App() {
  return (
    <main>
      <h2 data-testid="title-text">Jest 환경 설정</h2>
    </main>
  );
}

export default App;
