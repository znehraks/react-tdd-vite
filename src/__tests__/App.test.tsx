import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import App from "../App";

describe("app test", () => {
  test("jest title 테스트", () => {
    const AppComponent = render(<App />);
    expect(AppComponent).toMatchSnapshot();

    const titleText = screen.getByTestId("title-text");
    // expect(titleText).not.toBeNull();
    expect(titleText).toBeInTheDocument();
  });
});
