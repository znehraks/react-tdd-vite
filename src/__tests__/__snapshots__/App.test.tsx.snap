// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app test jest title 테스트 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <h2
          data-testid="title-text"
        >
          Jest 환경 설정
        </h2>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <h2
        data-testid="title-text"
      >
        Jest 환경 설정
      </h2>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
